Add DuckDB

### Package name

DuckDB

### Brief description of package

DuckDB is a modern alternative to SQLite designed to be fast in-process analytical database.

There is still a gitflow for building duckdb under Windows
https://github.com/duckdb/duckdb/blob/v0.10.2/.github/workflows/Windows.yml#L234

### URL for package's homepage

https://duckdb.org/

### Provide a basic test case to validate the package's functionality.

_No response_

### MINGW environments where you need the package

- [X] MINGW64

### Are you willing to submit a PR?

YES

Thank's @Biswa96! In `/mingw64/lib/cmake/DuckDB/DuckDBExports.cmake` at line `50` the installation prefix is hard coded with `/mingw64`. 
```
# The installation prefix configured by this project.
set(_IMPORT_PREFIX "/mingw64")
```

This leads to a cmake error when trying to look for the libs:

```
CMake Error at C:/msys64/mingw64/lib/cmake/DuckDB/DuckDBExports.cmake:124 (message):
  The imported target "duckdb" references the file

     "/mingw64/lib/libduckdb.dll.a"

  but this file does not exist.  Possible reasons include:
```

--- a/mingw-w64-duckdb/0001-duckdb-cmake-fix-lld-unknown-exclude-libs.patch
+++ b/mingw-w64-duckdb/0001-duckdb-cmake-fix-lld-unknown-exclude-libs.patch
@@ -1,0 +1,11 @@
+--- a/CMakeLists.txt
++++ b/CMakeLists.txt
+@@ -791,7 +791,7 @@
+   if(WASM_LOADABLE_EXTENSIONS)
+     set (CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -sSIDE_MODULE=1 -DWASM_LOADABLE_EXTENSIONS")
+   elseif (EXTENSION_STATIC_BUILD)
+-    if ("${CMAKE_CXX_COMPILER_ID}" STREQUAL "GNU" OR "${CMAKE_CXX_COMPILER_ID}" MATCHES "Clang")
++    if ("${CMAKE_CXX_COMPILER_ID}" STREQUAL "GNU" OR "${CMAKE_CXX_COMPILER_ID}" MATCHES "Clang" AND NOT WIN32)
+       if (APPLE)
+         set_target_properties(${TARGET_NAME} PROPERTIES CXX_VISIBILITY_PRESET hidden)
+         # Note that on MacOS we need to use the -exported_symbol whitelist feature due to a lack of -exclude-libs flag in mac's ld variant

--- a/mingw-w64-duckdb/0002-duckdb-cmake-fix-install-directory.patch
+++ b/mingw-w64-duckdb/0002-duckdb-cmake-fix-install-directory.patch
@@ -1,0 +1,11 @@
+--- a/CMakeLists.txt
++++ b/CMakeLists.txt
+@@ -57,7 +57,7 @@
+ set(INSTALL_INCLUDE_DIR
+     include
+     CACHE PATH "Installation directory for header files")
+-if(WIN32 AND NOT CYGWIN)
++if(WIN32 AND NOT CYGWIN AND NOT MINGW)
+   set(DEF_INSTALL_CMAKE_DIR cmake)
+ else()
+   set(DEF_INSTALL_CMAKE_DIR lib/cmake/DuckDB)

--- a/mingw-w64-duckdb/0003-duckdb-cmake-fix-import-prefix-relocation.patch
+++ b/mingw-w64-duckdb/0003-duckdb-cmake-fix-import-prefix-relocation.patch
@@ -1,0 +1,32 @@
+--- a/CMakeLists.txt
++++ b/CMakeLists.txt
+@@ -67,14 +67,6 @@
+     CACHE PATH "Installation directory for CMake files")
+ set(DUCKDB_EXPORT_SET "DuckDBExports")
+ 
+-# Make relative install paths absolute
+-foreach(p LIB BIN INCLUDE CMAKE)
+-  set(var INSTALL_${p}_DIR)
+-  if(NOT IS_ABSOLUTE "${${var}}")
+-    set(${var} "${CMAKE_INSTALL_PREFIX}/${${var}}")
+-  endif()
+-endforeach()
+-
+ # This option allows --gc-sections flag during extension linking to discard any unused functions or data
+ if (EXTENSION_STATIC_BUILD AND "${CMAKE_CXX_COMPILER_ID}" STREQUAL "GNU")
+   if ("${CMAKE_CXX_COMPILER_ID}" STREQUAL "GNU")
+@@ -1333,6 +1325,14 @@
+   configure_file(${CMAKE_CONFIG_TEMPLATE}
+                  "${PROJECT_BINARY_DIR}/DuckDBConfig.cmake" @ONLY)
+ 
++  # Make relative install paths absolute
++  foreach(p LIB BIN INCLUDE CMAKE)
++    set(var INSTALL_${p}_DIR)
++    if(NOT IS_ABSOLUTE "${${var}}")
++      set(${var} "${CMAKE_INSTALL_PREFIX}/${${var}}")
++    endif()
++  endforeach()
++
+   # Configure cmake package config for the install tree
+   file(RELATIVE_PATH REL_INCLUDE_DIR "${INSTALL_CMAKE_DIR}"
+        "${INSTALL_INCLUDE_DIR}")

--- a/mingw-w64-duckdb/PKGBUILD
+++ b/mingw-w64-duckdb/PKGBUILD
@@ -1,0 +1,74 @@
+# Maintainer: Biswapriyo Nath <nathbappai@gmail.com>
+
+_realname=duckdb
+pkgbase=mingw-w64-${_realname}
+pkgname=("${MINGW_PACKAGE_PREFIX}-${_realname}")
+pkgver=0.10.3
+pkgrel=1
+pkgdesc='An in-process SQL OLAP Database Management System (mingw-w64)'
+arch=('any')
+mingw_arch=('mingw64' 'ucrt64' 'clang64' 'clangarm64')
+url='https://duckdb.org/'
+license=('spdx:MIT')
+depepnds=(
+  "${MINGW_PACKAGE_PREFIX}-gcc-libs"
+  "${MINGW_PACKAGE_PREFIX}-libwinpthread"
+)
+makedepends=(
+  "${MINGW_PACKAGE_PREFIX}-cc"
+  "${MINGW_PACKAGE_PREFIX}-cmake"
+  "${MINGW_PACKAGE_PREFIX}-ninja"
+)
+source=("https://github.com/duckdb/duckdb/archive/v${pkgver}/${_realname}-v${pkgver}.tar.gz"
+        "0001-duckdb-cmake-fix-lld-unknown-exclude-libs.patch"
+        "0002-duckdb-cmake-fix-install-directory.patch")
+noextract=("${_realname}-v${pkgver}.tar.gz")
+sha256sums=('7855587b3491dd488993287caee28720bee43ae28e92e8f41ea4631e9afcbf88'
+            '8ff46f7426b402a102bb8bdbde4111f3af24fc8f5883e0128bd88fde6bb99243'
+            '332f22f3f8095b940d76c6224dd569b2fdc02a6a0b37f3b84e462d15af16da2e')
+
+_apply_patch_with_msg() {
+  for _patch in "$@"
+  do
+    msg2 "Applying ${_patch}"
+    patch -p1 -i "${srcdir}/${_patch}"
+  done
+}
+
+prepare() {
+  tar xf "${_realname}-v${pkgver}.tar.gz" || true
+
+  cd "${_realname}-${pkgver}"
+
+  _apply_patch_with_msg \
+    0001-duckdb-cmake-fix-lld-unknown-exclude-libs.patch \
+    0002-duckdb-cmake-fix-install-directory.patch
+}
+
+build() {
+  declare -a _extra_config
+  if check_option "debug" "n"; then
+    _extra_config+=("-DCMAKE_BUILD_TYPE=Release")
+  else
+    _extra_config+=("-DCMAKE_BUILD_TYPE=Debug")
+  fi
+
+  MSYS2_ARG_CONV_EXCL="-DCMAKE_INSTALL_PREFIX=" \
+    "${MINGW_PREFIX}"/bin/cmake.exe \
+      -GNinja \
+      -DCMAKE_INSTALL_PREFIX="${MINGW_PREFIX}" \
+      "${_extra_config[@]}" \
+      -DBUILD_SHARED_LIBS=ON \
+      -DBUILD_UNITTESTS=OFF \
+      -DOVERRIDE_GIT_DESCRIBE="v${pkgver}" \
+      -S "${_realname}-${pkgver}" \
+      -B "build-${MSYSTEM}"
+
+  "${MINGW_PREFIX}"/bin/cmake.exe --build "build-${MSYSTEM}"
+}
+
+package() {
+  DESTDIR="${pkgdir}" "${MINGW_PREFIX}"/bin/cmake.exe --install "build-${MSYSTEM}"
+
+  install -Dm644 "${srcdir}/${_realname}-${pkgver}/LICENSE" "${pkgdir}${MINGW_PREFIX}/share/licenses/${_realname}/LICENSE"
+}

--- a/mingw-w64-duckdb/PKGBUILD
+++ b/mingw-w64-duckdb/PKGBUILD
@@ -21,11 +21,13 @@ makedepends=(
 )
 source=("https://github.com/duckdb/duckdb/archive/v${pkgver}/${_realname}-v${pkgver}.tar.gz"
         "0001-duckdb-cmake-fix-lld-unknown-exclude-libs.patch"
-        "0002-duckdb-cmake-fix-install-directory.patch")
+        "0002-duckdb-cmake-fix-install-directory.patch"
+        "0003-duckdb-cmake-fix-import-prefix-relocation.patch")
 noextract=("${_realname}-v${pkgver}.tar.gz")
 sha256sums=('04e472e646f5cadd0a3f877a143610674b0d2bcf9f4102203ac3c3d02f1c5f26'
             '8ff46f7426b402a102bb8bdbde4111f3af24fc8f5883e0128bd88fde6bb99243'
-            '332f22f3f8095b940d76c6224dd569b2fdc02a6a0b37f3b84e462d15af16da2e')
+            '332f22f3f8095b940d76c6224dd569b2fdc02a6a0b37f3b84e462d15af16da2e'
+            '7bac703bb081460b519420d82af949b44ce5203f55973b11ddd87750e83fcaad')
 
 _apply_patch_with_msg() {
   for _patch in "$@"
@@ -42,7 +44,8 @@ prepare() {
 
   _apply_patch_with_msg \
     0001-duckdb-cmake-fix-lld-unknown-exclude-libs.patch \
-    0002-duckdb-cmake-fix-install-directory.patch
+    0002-duckdb-cmake-fix-install-directory.patch \
+    0003-duckdb-cmake-fix-import-prefix-relocation.patch
 }
 
 build() {

--- a/mingw-w64-duckdb/PKGBUILD
+++ b/mingw-w64-duckdb/PKGBUILD
@@ -3,7 +3,7 @@
 _realname=duckdb
 pkgbase=mingw-w64-${_realname}
 pkgname=("${MINGW_PACKAGE_PREFIX}-${_realname}")
-pkgver=0.10.3
+pkgver=1.0.0
 pkgrel=1
 pkgdesc='An in-process SQL OLAP Database Management System (mingw-w64)'
 arch=('any')
@@ -23,7 +23,7 @@ source=("https://github.com/duckdb/duckdb/archive/v${pkgver}/${_realname}-v${pkg
         "0001-duckdb-cmake-fix-lld-unknown-exclude-libs.patch"
         "0002-duckdb-cmake-fix-install-directory.patch")
 noextract=("${_realname}-v${pkgver}.tar.gz")
-sha256sums=('7855587b3491dd488993287caee28720bee43ae28e92e8f41ea4631e9afcbf88'
+sha256sums=('04e472e646f5cadd0a3f877a143610674b0d2bcf9f4102203ac3c3d02f1c5f26'
             '8ff46f7426b402a102bb8bdbde4111f3af24fc8f5883e0128bd88fde6bb99243'
             '332f22f3f8095b940d76c6224dd569b2fdc02a6a0b37f3b84e462d15af16da2e')
 

